demos/fleet_provisioning/fleet_provisioning_csr/fleet_provisioning_csr_demo.c

```diff
@@ -0,0 +1,1248 @@
+
+            responseStatus = ResponseAccepted;
+
+            ( void ) memcpy( ( void * ) payloadBuffer,
+                             ( const void * ) pPublishInfo->pPayload,
+                             ( size_t ) pPublishInfo->payloadLength );
+
+            payloadLength = pPublishInfo->payloadLength;
+        }
+        else if( api == FleetProvCborCreateCertFromCsrRejected )
+        {
+            LogError( ( "Fleet Provisioning CreateCertificateFromCsr rejected." ) );
+
+            responseStatus = ResponseRejected;
+        }
+        else if( api == FleetProvCborRegisterThingAccepted )
+        {
+            LogInfo( ( "Fleet Provisioning RegisterThing accepted." ) );
+
+            responseStatus = ResponseAccepted;
+
+            ( void ) memcpy( ( void * ) payloadBuffer,
+                             ( const void * ) pPublishInfo->pPayload,
+                             ( size_t ) pPublishInfo->payloadLength );
+
+            payloadLength = pPublishInfo->payloadLength;
+        }
+        else if( api == FleetProvCborCreateCertFromCsrRejected )
+        {
+            LogError( ( "Fleet Provisioning RegisterThing rejected." ) );
+
+            responseStatus = ResponseRejected;
+        }
+        else
+        {
+            LogError( ( "Unexpected Fleet Provisioning message. Topic: %.*s.",
+                        ( int ) pPublishInfo->topicNameLength,
+                        ( const char * ) pPublishInfo->pTopicName ) );
+        }
+    }
+}
+/*-----------------------------------------------------------*/
+
+static bool getCsr( char * buffer,
+                    size_t bufferLength,
+                    size_t * outCsrLength )
+{
+    FILE * file;
+    size_t length = 0;
+    bool status = true;
+
+    file = fopen( PROVISIONING_CSR_PATH, "rb" );
+
+    if( file == NULL )
+    {
+        LogError( ( "Error opening file at PROVISIONING_CSR_PATH." ) );
+        status = false;
+    }
+    else
+    {
+        int result;
+        result = fseek( file, 0L, SEEK_END );
+
+        if( result == -1 )
+        {
+            LogError( ( "Failed to seek CSR file." ) );
+            status = false;
+        }
+        else
+        {
+            long lenResult = -1;
+            lenResult = ftell( file );
+
+            if( lenResult == -1 )
+            {
+                LogError( ( "Failed to get length of CSR file." ) );
+                status = false;
+            }
+            else
+            {
+                length = ( size_t ) lenResult;
+            }
+        }
+
+        if( status == true )
+        {
+            if( length > bufferLength )
+            {
+                LogError( ( "Buffer too small for CSR" ) );
+                status = false;
+            }
+        }
+
+        if( status == true )
+        {
+            result = fseek( file, 0L, SEEK_SET );
+
+            if( result == -1 )
+            {
+                LogError( ( "Failed to seek CSR file." ) );
```

Comment:
"seek CSR file" is not a helpful log message... Can you update the log message to be more expressive about the problem?